client/shaders/object_shader/opengl_fragment.glsl

```diff
@@ -164,5 +164,12 @@ void main(void)
 		- fogShadingParameter * length(eyeVec) / fogDistance, 0.0, 1.0);
 	col = mix(skyBgColor, col, clarity);
 
-	gl_FragColor = vec4(col.rgb, base.a);
+	vec4 final_col = vec4(col.rgb, base.a);
+#ifdef SECONDSTAGE
+	gl_FragData[0] = final_col;
+	gl_FragData[1] = vec4((vNormal+1)/2, DRAW_TYPE);
+	gl_FragData[2] = vec4((fogDistance-length(eyeVec))/1000, 0,0,0);
```

Comment:
Again, why not `length(eyeVec)`?

Reply:
This code clips the depth from 1 (closest) to 0 (farthest). It basically uses the same code as the fog.

Reply:
I see; my question is, is that necessary? I don’t remember are values clipped to [0, 1] if the destination buffer doesn’t require that.

Reply:
I think the fog and this code calculate the distance and not the depth. See #4628.